### PR TITLE
drivers: bmx280: adapt to new I2C api.

### PR DESCRIPTION
**Pull request by basilfx**
### Contribution description

This PR updates the BMx280 driver. Tested this one on the SLTB001A (EFM32, adapted).

It's probably better to wait for #8383, but it was easy to adapt for the time being.

### Issues/PRs references

#6577

### CODE REVIEW

File: drivers/bmx280/bmx280.c [hunk 1]
```diff
@@ -225,13 +219,15 @@ static int read_calibration_data(bmx280_t* dev)
...
 
     memset(buffer, 0, sizeof(buffer));
-    nr_bytes = i2c_read_regs(dev->params.i2c_dev, dev->params.i2c_addr, offset,
-                             buffer, nr_bytes_to_read);
-    if (nr_bytes != nr_bytes_to_read) {
+    i2c_acquire(dev->params.i2c_dev);
+    result = i2c_read_regs(dev->params.i2c_dev, dev->params.i2c_addr, offset,
+                           buffer, nr_bytes_to_read, 0);
+    i2c_release(dev->params.i2c_dev);
+    if (result != I2C_ACK) {
```
Comment (aabadie) [inline drivers/bmx280/bmx280.c]: `I2C_ACK` 'return code was removed from the API, prefer a check like `!=` or `<0`.

File: drivers/bmx280/bmx280.c [hunk 2]
```diff
@@ -315,17 +313,19 @@ static int do_measurement(const bmx280_t* dev)
...
     uint8_t offset = BMX280_PRESSURE_MSB_REG;
 
-    nr_bytes = i2c_read_regs(dev->params.i2c_dev, dev->params.i2c_addr,
-                             offset, measurement_regs, nr_bytes_to_read);
-    if (nr_bytes != nr_bytes_to_read) {
+    i2c_acquire(dev->params.i2c_dev);
+    result = i2c_read_regs(dev->params.i2c_dev, dev->params.i2c_addr,
+                           offset, measurement_regs, nr_bytes_to_read, 0);
+    i2c_release(dev->params.i2c_dev);
+    if (result != I2C_ACK) {
```
Comment (aabadie) [inline drivers/bmx280/bmx280.c]: Same here.

**Comment by keestux** [comment]
@basilfx can you rebase onto the latest new_i2c_if?

**Comment by basilfx** [comment]
Sorry for the delay. I have rebased and adapted, but I cannot test this one since I have to fix EFM32 first (which will take some time).

**Comment by keestux** [comment]
Thanks
I've tested it with SODAQ Autonomo, and I confirm it is working

**Comment by basilfx** [comment]
@dylad Can you tell me which lines? Note that I only adapted this driver to make it work with the new interface, so most code was already in place.

**Comment by basilfx** [comment]
Squashed.

**Comment by dylad** [comment]
@aabadie Can we dismiss your review ?

**Pull request merged**